File: LeetCode_01/Solution_ZhongJing.kt
package LeetCode_01

/**
 * 日期：2023-07-02
 * 作者：仲景
 */
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val m = mutableMapOf<Int, Int>()
        nums.forEachIndexed { index, num ->
            val curTarget = target - num
            m[curTarget]?.let { return intArrayOf(index, it) }
            m[num] = index
        }
        return intArrayOf()
    }
}

File: LeetCode_01/Solution_ZhongJing.py
from typing import List


class Solution:
    """
    题目：1.两数之和
    日期：2023-08-04
    作者：仲景
    """
    
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # 创建一个map
        numMap = {}
        for i in range(len(nums)):
            cur = nums[i]
            if target - cur in numMap:
                return [i, numMap[target - cur]]
            numMap[cur] = i